contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,491 @@
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's donated amount of sOHM at current index
+        @param depositId_ Deposit id for this donation
+    */
+    function redeemYieldAsSohm(uint256 depositId_) external {
+        uint256 amountRedeemed = _redeem(depositId_);
+
+        IERC20(sOHM).approve(address(staking), amountRedeemed);
+        staking.unwrap(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's full donated amount of sOHM at current index as gOHM
+    */
+    function redeemAllYield() external {
+        uint256 amountRedeemed = _redeemAll();
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's full donated amount of sOHM at current index as gOHM
+    */
+    function redeemAllYieldAsSohm() external {
+        uint256 amountRedeemed = _redeemAll();
+        
+        IERC20(sOHM).approve(address(staking), amountRedeemed);
+        staking.unwrap(msg.sender, amountRedeemed);
+    }
+
+    /************************
+     * Internal Functions
+     ************************/
+
+    /**
+        @notice Creates a new deposit directing the yield from the deposited gOHM amount
+                to the prescribed recipient
+        @param amount_ Quantity of gOHM deposited redirecting yield to the recipient
+        @param recipient_ The address of the user who will be entitled to claim the donated yield
+    */
+    function _createDeposit(uint256 amount_, address recipient_) internal returns (uint256 depositId) {
+        depositId = _deposit(msg.sender, amount_);
+        recipientIds[recipient_].push(depositId);
+        recipientLookup[depositId] = recipient_;
+
+        emit Deposited(msg.sender, recipient_, amount_);
+    }
+
+    /**
+        @notice Increases the amount of gOHM directing yield to a recipient
+        @param depositId_ The global ID number of the deposit to add the additional deposit to
+        @param amount_ Quantity of new gOHM deposited redirecting yield to the current deposit's recipient
+    */
+    function _increaseDeposit(uint256 depositId_, uint256 amount_) internal {
+        _addToDeposit(depositId_, amount_);
+
+        emit Deposited(depositInfo[depositId_].depositor, recipientLookup[depositId_], amount_);
+    }
+
+    /**
+        @notice Withdraw gOHM deposit from vault
+        @param depositId_ Deposit ID to remove gOHM deposit from
+        @param amount_ Amount of gOHM deposit to remove and return to donor 
+    */
+    function _withdraw(uint256 depositId_, uint256 amount_) internal returns (uint256 amountWithdrawn) {
+        if (isInvalidWithdrawal(amount_)) revert YieldDirector_InvalidWithdrawal();
+
+        DepositInfo storage currDeposit = depositInfo[depositId_];
+        uint256 withdrawableBalance = _toAgnostic(currDeposit.principalAmount);
+
+        if (amount_ >= withdrawableBalance) {
```

Comment:
how bad would it be to just instead of making withdrawableBalance by storing 2 variables just go 
depositInfo[depositId_]. principalAmount ?

Seems like that might save some gas since those 2 variables are only used once.

Reply:
so change it to `if (amount_ >= _toAgnostic(depositInfo[depositId_].principalAmount)` ?